lab4/src/main/java/agh/ics/oop/RectangularMap.java

```diff
@@ -0,0 +1,59 @@
+package agh.ics.oop;
+
+import java.util.HashMap;
+
+class RectangularMap implements IWorldMap {
+
+    // hashmap which keys are position (as vector) and values are Animal
+    private final HashMap<Vector2d, Animal> animalPositions;
+
+    private final Vector2d UPPER_RIGHT;
```

Comment:
dużymi literami piszemy tylko stałe `static final`, a to jest zwykły atrybut - każda mapa ma swój własny `upperRight`